### ISSUE TITLE
ROA using 4294967295 32-bit last ASN for Negative Attestation

### ISSUE THREAD

**Issue by guzzim** [irrdnet/irrd#535]
**Describe the bug**
32-bit 4294967295 ASN used for Negative Attestation appear to cause the RPKI validation to fail on import of the json file.

Error message in the log is
2021-08-05 05:52:49,741 irrd[8971]: [irrd.mirroring.mirror_runners_import#INFO] Running full ROA import from: https://rpki.gin.ntt.net/api/export.json, SLURM None
2021-08-05 05:53:37,635 irrd[8971]: [irrd.rpki.importer#ERROR] Invalid value in ROA or SLURM: Invalid AS number 4294967295: valid range is 0-4294967295
2021-08-05 05:53:37,635 irrd[8971]: [irrd.mirroring.mirror_runners_import#ERROR] An exception occurred while attempting a ROA import: Invalid value in ROA or SLURM: Invalid AS number 4294967295: valid range is 0-4294967295

from the NTT RPKI export.json
(irrd-venv) bash-4.4$ cat /tmp/tmpovt_64u4 | grep "103.138.210.0/24"
		{ "asn": 0, "prefix": "103.138.210.0/24", "maxLength": 24, "ta": "apnic", "expires": 1628276597 },
		{ "asn": 65534, "prefix": "103.138.210.0/24", "maxLength": 24, "ta": "apnic", "expires": 1628276597 },
		{ "asn": 139038, "prefix": "103.138.210.0/24", "maxLength": 24, "ta": "apnic", "expires": 1628276597 },
		{ "asn": 4200000000, "prefix": "103.138.210.0/24", "maxLength": 24, "ta": "apnic", "expires": 1628276597 },
		{ "asn": 4294967295, "prefix": "103.138.210.0/24", "maxLength": 24, "ta": "apnic", "expires": 1628276597 },

irrd=# select count(*) from rpsl_objects where rpki_status = 'not_found';
-[ RECORD 1 ]---
count | 10735821

irrd=# select count(*) from rpsl_objects;
-[ RECORD 1 ]---
count | 10735821

**To Reproduce**
An 32-bit ASN equal to 4294967295 in the ROA.

**Expected behaviour**
Not entirely sure myself, this is new ground for me; RFC6483/7607 I thought was supposed to use the very first ASN value for this purpose.

**IRRd version you are running**
(irrd-venv) bash-4.4$ whois -h localhost '!v'
A25
IRRd -- version 4.2.0rc1
C

**Additional context**

**Comment by guzzim** [comment]
This got me out of trouble with my sandbox

```
diff --git a/irrd/utils/validators.py b/irrd/utils/validators.py
index 7dcece3..3cee2aa 100644
--- a/irrd/utils/validators.py
+++ b/irrd/utils/validators.py
@@ -19,7 +19,7 @@ def parse_as_number(value: Union[str, int], permit_plain=False) -> Tuple[str, in
     else:
         value_int = value
 
-    if not 0 <= value_int < 4294967295:
+    if not 0 <= value_int < 4294967296:
         raise ValidationError(f'Invalid AS number {value}: valid range is 0-4294967295')
```

It might be more preferable the script skips over the dud entries and logs them as opposed to aborting all together.

**Comment by mxsasha** [comment]
You are entirely correct, the ASN validator is off by one. Your proposed patch is also correct, and also fixes this in other locations. I'll pick it up the bugfix here.

> It might be more preferable the script skips over the dud entries and logs them as opposed to aborting all together.

I am personally leaning to either having a valid file or rejecting it. In this case, it is an error in our parser. However, in general, the entire file should be valid. A file that contains invalid entries should not be trusted, because it means other entries may also be incorrect even if they are syntactically valid.

**Comment by guzzim** [comment]
Can't argue with that, I think you're spot on there.

Many thanks

**Comment by troy2914** [comment]
This also affects IRRd -- version 4.1.4

**Comment by mxsasha** [comment]
Reopening for backport to 4.1.

**Comment by mxsasha** [comment]
Released in 4.1.5
